src/main/java/org/breedinginsight/brapi/v2/BrAPIProgramsController.java

```diff
@@ -139,6 +151,14 @@ public HttpResponse<BrAPIProgramListResponse> programsGet(@PathVariable("program
                                     @QueryValue("page") Optional<Integer> page,
                                     @QueryValue("pageSize") Optional<Integer> pageSize) {
 
+        //If programId supplied, check if program exists
+        if (programId != null) {
+            Optional<Program> optProgram = programService.getById(programId);
```

Comment:
I think this optional could be reused below to get the program rather than calling programService.getById again. Also, if programId were null the programService.getById below would be called with the null programId based on the current check.

Reply:
Good catch, fixed